LanguageFeatures/Static-access-shorthand/constant_expression_A10_t17.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A static member shorthand expression should be a potentially
+/// constant expression if the corresponding explicit static member plus
+/// selectors expression would be, which currently means that it's a potentially
+/// constant expression if and only if it's a constant expression.
+///
+/// @description Checks that it is a compile-time error if a non-constant static
+/// member shorthand expression is used in a constant `e1 ? e2 : e3` expression.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=enum-shorthands
+
+class C {
+  static C one = const C(1);
+  static const C two = const C(2);
```

Comment:
A constant variable and a non-constant one? Swapped later? Great, makes the tests very focused and still not too repetitive!